### PR TITLE
feat(pipes-targets): add API Gateway

### PR DESCRIPTION
**Pull request by msambol**
Add API Gateway REST API as a Pipes target.

Original PR with community review from @nmussy: https://github.com/aws/aws-cdk/pull/30772

### CODE REVIEW

**Comment by msambol** [comment]
@Leo10Gama Got another one for you :) This was first reviewed in https://github.com/aws/aws-cdk/pull/31589, but was closed because of inactivity/merge conflicts.

File: packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-gateway.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  logGroupName: fn.logGroup.logGroupName,
+  limit: 3,
+});
+
+const message = putMessageOnQueue.next(logEvents);
+
+// Check that the Lambda was invoked successfully from API GW.
+// Checking for the actual payload from SQS results in the following:
+// "Response object is too long." The assertion below checks that
+// API GW invoked the function.
```
Comment (Leo10Gama) [inline packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-gateway.ts:82]: Curious as to why the response object is too long for this test. The strategy seems the same as other targets, which can be read just fine, is this just the nature of the API gateway logs?

**Review by Leo10Gama — COMMENTED** [review]
Looks pretty good! Because of some of the policy permissions, I'm adding in the `needs-security-review` label as well, and review with security to verify the changes are OK. Other than that, one comment about how we're verifying the integ test.

File: packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-gateway.ts (hunk 1 above)
Comment (msambol) [reply packages/@aws-cdk/aws-pipes-targets-alpha/test/integ.api-gateway.ts:82]: Yeah, it logs the entire payload. But I did manually check and the message from SQS made it through to the Lambda.

**Review by Leo10Gama — DISMISSED** [review]
Changes look good, thanks again for the contribution!

**Comment by Leo10Gama** [comment]
@mergifyio update

**Comment by msambol** [comment]
@Leo10Gama mind re-approving this one? looks like the merge dismissed your review.

**Review by Leo10Gama — APPROVED** [review]

**Pull request merged**